src/logical/program_analysis/analysis.rs

```diff
@@ -507,6 +509,13 @@ impl Program {
                     .get(left_variable)
                     .expect("Previous analysis should have assigned a type to each variable.");
 
+                if filter.operation != FilterOperation::Equals
+                    && *variable_type != LogicalTypeEnum::Double
+                    && *variable_type != LogicalTypeEnum::UnsignedInteger
```

Comment:
I feel like the information which logical types support comparisons should be closer to their definition. Otherwise, I'm afraid that we will lose track of all the places that need adjustments when new types are added.
Maybe one could just have a method on the enum that returns a `bool` indicating if comparsion is supported.

Reply:
Yes this makes sense